base/broadcast.jl

```diff
@@ -961,28 +947,12 @@ end
 
 ## Tuple methods
 
-@inline copy(bc::Broadcasted{Style{Tuple}}) =
-    tuplebroadcast(longest_tuple(nothing, bc.args), bc)
-@inline tuplebroadcast(::NTuple{N,Any}, bc) where {N} = ntuple(k -> @inbounds(_broadcast_getindex(bc, k)), Val(N))
-# This is a little tricky: find the longest tuple (first arg) within the list of arguments (second arg)
-# Start with nothing as a placeholder and go until we find the first tuple in the argument list
-longest_tuple(::Nothing, t::Tuple{Tuple,Vararg{Any}}) = longest_tuple(t[1], tail(t))
-# Or recurse through nested broadcast expressions
-longest_tuple(::Nothing, t::Tuple{Broadcasted,Vararg{Any}}) = longest_tuple(longest_tuple(nothing, t[1].args), tail(t))
-longest_tuple(::Nothing, t::Tuple) = longest_tuple(nothing, tail(t))
-# And then compare it against all other tuples we find in the argument list or nested broadcasts
-longest_tuple(l::Tuple, t::Tuple{Tuple,Vararg{Any}}) = longest_tuple(_longest_tuple(l, t[1]), tail(t))
-longest_tuple(l::Tuple, t::Tuple) = longest_tuple(l, tail(t))
-longest_tuple(l::Tuple, ::Tuple{}) = l
-longest_tuple(l::Tuple, t::Tuple{Broadcasted}) = longest_tuple(l, t[1].args)
-longest_tuple(l::Tuple, t::Tuple{Broadcasted,Vararg{Any}}) = longest_tuple(longest_tuple(l, t[1].args), tail(t))
-# Support only 1-tuples and N-tuples where there are no conflicts in N
-_longest_tuple(A::Tuple{Any}, B::Tuple{Any}) = A
-_longest_tuple(A::Tuple{Any}, B::NTuple{N,Any}) where N = B
-_longest_tuple(A::NTuple{N,Any}, B::Tuple{Any}) where N = A
-_longest_tuple(A::NTuple{N,Any}, B::NTuple{N,Any}) where N = A
-@noinline _longest_tuple(A, B) =
-    throw(DimensionMismatch("tuples $A and $B could not be broadcast to a common size"))
+@inline function copy(bc::Broadcasted{Style{Tuple}})
+    axes = broadcast_axes(bc)
+    length(axes) == 1 || throw(DimensionMismatch("tuple only supports one dimension"))
+    N = Val(length(axes[1]))
```

Comment:
Whoa, is this actually inferable!?

Reply:
Do we even need the `Val` anymore?

```julia
julia> foo(x::AbstractArray) = ntuple(d->1, ndims(x))
foo (generic function with 1 method)

julia> foo(rand(2,2))
(1, 1)

julia> @code_warntype foo(rand(2,2))
Body::Tuple{Int64,Int64}
1 1 ─     return (1, 1)                                                                                                                                                                                                                  │
```

Reply:
`broadcast_axes` is a constant tuple (`Const(tuple(OneTo(n),))`), so yes, it's pretty easy to infer through this now

We don't need the `Val`, but it currently still can give better code results in the fully-inferred case.